tests/conftest.py

```diff
@@ -205,3 +207,29 @@ def single_version():
 @pytest.fixture(scope="function")
 def no_numpy_version():
     return {"python": ["2.7.*", "3.5.*"]}
+
+
+@pytest.fixture(scope="function")
+def variants_conda_build_sysroot():
+    if not on_mac:
+        return {}
+
+    return {
+        "CONDA_BUILD_SYSROOT": [
+            subprocess.run(
+                ["xcrun", "--sdk", "macosx", "--show-sdk-path"],
+                check=True,
+                capture_output=True,
+                text=True,
+            ).stdout.strip()
+        ],
+        "MACOSX_DEPLOYMENT_TARGET": [
+            subprocess.run(
+                ["xcrun", "--sdk", "macosx", "--show-sdk-version"],
+                check=True,
+                capture_output=True,
+                text=True,
+            ).stdout.strip(),
+            "10.9",
+        ],
+    }
```

Comment:
For the builds at conda-forge we usually have them as environment variables and then override them via cbc.yaml is required.
To be closer to that, we could also have the fixture like this:
```suggestion
def variants_conda_build_sysroot(monkeypatch):
    if not on_mac:
        return {}

    monkeypatch.setenv(
        "MACOSX_DEPLOYMENT_TARGET",
        subprocess.run(
            ["xcrun", "--sdk", "macosx", "--show-sdk-version"],
            check=True,
            capture_output=True,
            text=True,
        ).stdout.strip(),
    )
    monkeypatch.setenv(
        "CONDA_BUILD_SYSROOT",
        subprocess.run(
            ["xcrun", "--sdk", "macosx", "--show-sdk-path"],
            check=True,
            capture_output=True,
            text=True,
        ).stdout.strip(),
    )
    return {"MACOSX_DEPLOYMENT_TARGET": ["10.9"]}
```

Reply:
@mbargull the intention was for `{"MACOSX_DEPLOYMENT_TARGET": ["10.9"]}` to always override the `MACOSX_DEPLOYMENT_TARGET` environment variable (assuming its passed to `api.build`), right?

Reply:
Ah, whoops, my bad. No, I meant to still have it run two versions one without the override and one with. Hold on...